ocf_datapipes/select/select_time_periods.py

```diff
@@ -0,0 +1,25 @@
+from typing import Union
+
+import pandas as pd
+import xarray as xr
+from torchdata.datapipes import functional_datapipe
+from torchdata.datapipes.iter import IterDataPipe
+
+
+@functional_datapipe("select_time_periods")
+class SelectTimePeriodsIterDataPipe(IterDataPipe):
+    def __init__(
+        self, source_dp: IterDataPipe, time_periods: pd.DataFrame, dim_name: str = "time_utc"
+    ):
+        self.source_dp = source_dp
+        self.time_periods = time_periods
```

Comment:
Good to descirbe exactly what columnds are in time_periods, i think it 'start_dt' and 'end_dt'

Could this be start_datetime and end_datetime? i dont like the abbreviation when it not really needed. dt could be deltatime